src/commands/command.cr

```diff
@@ -73,8 +74,30 @@ module Shards
       Shards::Lock.write(packages, override_path, LOCK_FILENAME)
     end
 
-    def handle_resolver_errors(&)
+    private def log_available_tags(conflicts)
+      conflicts.join(separator: "\n") do |k, v|
+        req = v.requirement
+        tags = req.resolver.available_tags
+        req = req.requirement
+
+        if req.is_a?(Version) || (req.is_a?(VersionReq) && req.patterns.size == 1 && req.patterns[0] !~ /^(<|>|=)/)
+          req = "v" + req.to_s
+          found = Levenshtein.find(req, tags, 6)
+          "For #{k} the closest available tag to #{req} is: #{found}"
+        elsif tags.empty?
+          "#{k} doesn't have any tag"
+        else
+          "For #{k} the last available tags are #{tags.reverse.first(5).join(", ")}"
+        end
+      end
+    end
+
+    def handle_resolver_errors(solver, &)
       yield
+    rescue e : Molinillo::VersionConflict(Shards::Dependency, Shards::Spec)
+      Log.error { e.message }
+      Log.error { log_available_tags(e.conflicts) }
```

Comment:
**polish:** This should probably be a single error message. The conflict information is just context for the version conflict error, not itself another error.

Reply:
If something depends on the full exact text of the error message, it should be expected to break.
We cannot possible guarantee stability on that.